GNOME: 40 → 41

jtojnar opened the pull request:
https://discourse.gnome.org/t/gnome-41-released/7603/1

cc @NixOS/gnome 

https://github.com/NixOS/nixpkgs/projects/41
https://hydra.nixos.org/jobset/nixpkgs/gnome

Commit:
glibmm: 2.64.5 -> 2.66.1

- https://gitlab.gnome.org/GNOME/glibmm/-/blob/2.66.1/NEWS

Commit:
cairomm: 1.12.2 -> 1.14.3

Commit:
pangomm: 2.42.2 -> 2.46.1

- https://gitlab.gnome.org/GNOME/pangomm/-/blob/2.46.1/NEWS

jtojnar committed:
gtk4: 4.2.1 → 4.4.0

https://gitlab.gnome.org/GNOME/gtk/-/blob/4.4.0/NEWS

https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/3391
- ffmpeg is experimental (no audio support), let’s just use GStreamer
- vulkan is experimental and can cause crashes in inspector, let’s disable it by default

jtojnar committed:
gi-docgen: 2021.5 → 2021.7

https://gitlab.gnome.org/GNOME/gi-docgen/-/tags/2021.6
https://gitlab.gnome.org/GNOME/gi-docgen/-/tags/2021.7

mweinelt commented on pkgs/development/libraries/gtk/4.x.nix:
```diff
@@ -37,6 +33,7 @@
 , x11Support ? stdenv.isLinux
 , waylandSupport ? stdenv.isLinux
 , libGL
+, vulkanSupport ? false
```
Why is `vulkanSupport` disabled by default, when it was previously enable when not on Darwin?

bobby285271 replied:
https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/3391/diffs#ca9bb7eff80503c97c83505e8acea4002fd87ac6_58_54 it is experimental (and can cause crashes in inspector, according to the commit message from jtojnar).

mweinelt replied:
Worth a comment I guess.

jtojnar committed:
gnome.sushi: 3.38.1 → 41.0

jtojnar committed:
gnome.cheese: 3.38.0 → 41.0

jtojnar committed:
gnome.polari: 3.38.0 → 41.0

UI uses GTK 4, thumbnailer GTK 3 (due to a dependency on WebkitGtk, where GTK 4 support is still experimental).

Both work, though the console is full of warnings.

jtojnar committed:
pango: 1.48.5 → 1.48.10

https://gitlab.gnome.org/GNOME/pango/-/blob/1.48.10/NEWS

Commit:
vala_0_54: init at 0.54.1

- https://gitlab.gnome.org/GNOME/vala/blob/0.54.0/NEWS
- https://gitlab.gnome.org/GNOME/vala/blob/0.54.1/NEWS

Commit:
vala: default to vala_0_54

Commit:
at-spi2-core: 2.40.3 -> 2.42.0

- https://gitlab.gnome.org/GNOME/at-spi2-core/-/blob/AT_SPI2_CORE_2_42_0/NEWS

Commit:
gjs: 1.68.3 -> 1.70.0

- https://gitlab.gnome.org/GNOME/gjs/-/blob/1.70.0/NEWS

Commit:
glib: 2.68.4 -> 2.70.0

- https://gitlab.gnome.org/GNOME/glib/-/blob/2.70.0/NEWS

Commit:
glib-networking: 2.68.2 -> 2.70.0

- https://gitlab.gnome.org/GNOME/glib-networking/-/blob/2.70.0/NEWS

Commit:
libsoup: 2.72.0 -> 2.74.0

- https://gitlab.gnome.org/GNOME/libsoup/-/blob/2.74.0/NEWS

Commit:
libgudev: 236 -> 237

- https://gitlab.gnome.org/GNOME/libgudev/-/blob/237/NEWS

Commit:
libdazzle: 3.40.0 -> 3.42.0

- https://gitlab.gnome.org/GNOME/libdazzle/-/blob/3.42.0/NEWS

Commit:
librsvg: 2.50.7 -> 2.52.0

- https://gitlab.gnome.org/GNOME/librsvg/-/blob/2.52.0/NEWS

Commit:
gobject-introspection: 1.68.0 -> 1.70.0

- https://gitlab.gnome.org/GNOME/gobject-introspection/-/blob/1.70.0/NEWS

Commit:
gtksourceview4: 4.8.1 -> 4.8.2

- https://gitlab.gnome.org/GNOME/gtksourceview/-/blob/4.8.2/NEWS

Commit:
libmediaart: 1.9.4 -> 1.9.5

- https://gitlab.gnome.org/GNOME/libmediaart/-/blob/1.9.5/NEWS

Commit:
gsound: 1.0.2 -> 1.0.3

Commit:
gnome.gnome-boxes: 40.3 -> 41.1

- https://gitlab.gnome.org/GNOME/gnome-boxes/-/blob/41.1/NEWS

Commit:
libblockdev: 2.25 -> 2.26

- https://github.com/storaged-project/libblockdev/releases/tag/2.26-1
- https://github.com/storaged-project/libblockdev/blob/2.26-1/NEWS.rst

maxeaubrey commented:
`libvirt-glib` doesn't build with glib > 2.69 on the latest release. Reading through commits doesn't look like it's been addressed yet, and after a quick search through the mailing list I didn't see any patches either (not that I was super thorough).

Commit:
gnome.gnome-calendar: 40.2 -> 41.0

- https://gitlab.gnome.org/GNOME/gnome-calendar/-/blob/41.0/NEWS

Commit:
gnome.gnome-desktop: 40.4 -> 41.0

- https://gitlab.gnome.org/GNOME/gnome-desktop/-/blob/41.0/NEWS

Commit:
gnome.gnome-music: 40.1.1 -> 41.0

- https://gitlab.gnome.org/GNOME/gnome-music/-/blob/41.0/NEWS

Commit:
gnome.gnome-characters: 40.0 -> 41.0

- https://gitlab.gnome.org/GNOME/gnome-characters/-/blob/41.0/NEWS

Commit:
gnome.gnome-todo: 40.0 -> 41.0

- https://gitlab.gnome.org/GNOME/gnome-todo/-/blob/41.0/NEWS

Commit:
gnome.gnome-weather: 40.1 -> 41.0

- https://gitlab.gnome.org/GNOME/gnome-weather/-/blob/41.0/NEWS

bobby285271 commented:
Some probably off-topic note: We just pinned Vala to 0.52 for Pantheon packages that cannot build with 0.54 [in this PR](https://github.com/NixOS/nixpkgs/pull/139163).

> `libvirt-glib` doesn't build with glib > 2.69 on the latest release.

A newbie question: I tried building `libvirt-glib` on this PR and it failed with `G_TYPE_IS_FINAL` not available before 2.70 ([full log](https://gist.github.com/bobby285271/0b896ace8b941dce5b29180227108c2b)). Will `mesonFlags = [ "-Dc_args=-DGLIB_DISABLE_DEPRECATION_WARNINGS" ];` *work* in this case (it does successfully *compile* for me)?

Not sure if these are related: (1) [`GLIB_VERSION_MAX_ALLOWED` set in `libvirt-glib`](https://github.com/libvirt/libvirt-glib/blob/9a34c4ea55e0246c34896e48b8ecd637bc559ac7/meson.build#L521) (2) [Fedora Rawhide is building this with `glib` 2.69.2 +  `gobject-introspection` 1.68.0](https://kojipkgs.fedoraproject.org//packages/libvirt-glib/4.0.0/3.fc36/data/logs/aarch64/build.log) (3) [Add "final" class attribute merge request in `gobject-introspection`](https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/257) (4) [Add "final" type flag merge request in `glib`](https://gitlab.gnome.org/GNOME/glib/-/merge_requests/1937).

SuperSandro2000 commented on pkgs/desktops/gnome/core/gnome-desktop/bubblewrap-paths.patch:
```diff
@@ -1,15 +1,17 @@
+diff --git a/libgnome-desktop/gnome-desktop-thumbnail-script.c b/libgnome-desktop/gnome-desktop-thumbnail-script.c
+index ddcc1511..546c2a36 100644
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/libraries/gtk/4.x.nix:
```diff
@@ -86,27 +83,25 @@ stdenv.mkDerivation rec {
     python3
     sassc
     gi-docgen
+    libxml2 # for xmllint
```
Should we make an extra output for xmllint?

jtojnar replied:
Unfortunately, that currently is not really possible to do conveniently – build systems generally support only single `bindir` per project. (Eventually, I want to have Debian-level granularity packages but that would require massive redesign of Meson and porting the affected projects to Meson.) Though, I wonder if we should explicitly list the outputs used – it used to be that way in the past.

SuperSandro2000 commented on pkgs/development/libraries/libsoup/default.nix:
```diff
@@ -23,21 +23,13 @@
   };
 
-  patches = [
-    (fetchpatch {
```
Please remove the unused fetchpatch from inputs.

jtojnar committed:
libvirt-glib: fix build with GLib 2.70

jtojnar committed:
python3.pkgs.pygobject3: 3.40.1 → 3.42.0

https://gitlab.gnome.org/GNOME/pygobject/-/blob/3.42.0/NEWS

jtojnar committed:
libsoup: clean up

To keep in sync with libsoup3.

jtojnar committed:
libsoup3: init at 3.0.1

https://gitlab.gnome.org/GNOME/libsoup/-/blob/9be41160825b8238b482208b46bfe16b2f239ce3/NEWS

jtojnar committed:
webkitgtk: 2.32.3 → 2.34.0

Changes:
- https://webkitgtk.org/2021/05/14/webkitgtk2.33.1-released.html
- https://webkitgtk.org/2021/06/08/webkitgtk2.33.2-released.html
- https://webkitgtk.org/2021/08/16/webkitgtk2.33.3-released.html
- https://webkitgtk.org/2021/09/02/webkitgtk2.33.90-released.html
- https://webkitgtk.org/2021/09/17/webkitgtk2.33.91-released.html
- https://webkitgtk.org/2021/09/22/webkitgtk2.34.0-released.html

Notably, libsoup3 is used by default. We revert that to reduce breakage
so programs using libsoup 3.x will need to override the argument.

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -17541,6 +17541,8 @@ with pkgs;
 
   libsoup = callPackage ../development/libraries/libsoup { };
 
+  libsoup3 = callPackage ../development/libraries/libsoup/3.x.nix { };
```
```suggestion
  libsoup_3 = callPackage ../development/libraries/libsoup/3.nix { };
```

SuperSandro2000 commented on pkgs/development/libraries/libsoup/3.x.nix:
```diff
@@ -0,0 +1,102 @@
+  PKG_CONFIG_PATH = "${libnghttp2.dev}/lib/pkgconfig";
+
+  # HSTS tests fail.
+  doCheck = false;
```
Can we disable only this test?

jtojnar replied:
Not without patching those test declarations out. https://github.com/mesonbuild/meson/issues/6999

SuperSandro2000 commented on pkgs/development/libraries/libsoup/3.x.nix:
```diff
@@ -0,0 +1,102 @@
+, sqlite
+, glib-networking
+, gobject-introspection
+, withIntrospection ? stdenv.buildPlatform == stdenv.hostPlatform
+, vala
+, withVala ? stdenv.buildPlatform == stdenv.hostPlatform
```
Shouldn't this be one input called crossCompile or similar?

jtojnar replied:
Previously, `stdenv.isCross` was used but the cross people switched to this.

jtojnar committed:
gnome.adwaita-icon-theme: 40.1.1 → 41.0

https://ftp.gnome.org/pub/GNOME/sources/adwaita-icon-theme/41/adwaita-icon-theme-41.0.news

jtojnar committed:
gnome.baobab: 40.0 → 41.0

https://ftp.gnome.org/pub/GNOME/sources/baobab/41/baobab-41.0.news

jtojnar committed:
gnome.devhelp: 40.1 → 41.1

https://ftp.gnome.org/pub/GNOME/sources/devhelp/41/devhelp-41.alpha.news
https://ftp.gnome.org/pub/GNOME/sources/devhelp/41/devhelp-41.beta.news
https://ftp.gnome.org/pub/GNOME/sources/devhelp/41/devhelp-41.rc.news
https://ftp.gnome.org/pub/GNOME/sources/devhelp/41/devhelp-41.0.news
https://ftp.gnome.org/pub/GNOME/sources/devhelp/41/devhelp-41.1.news

gnome.devhelp: 40.1 → 41.2

jtojnar committed:
gnome.eog: 40.3 → 41.0

https://ftp.gnome.org/pub/GNOME/sources/eog/41/eog-41.beta.news
https://ftp.gnome.org/pub/GNOME/sources/eog/41/eog-41.rc.news
https://ftp.gnome.org/pub/GNOME/sources/eog/41/eog-41.0.news

jtojnar committed:
gnome.epiphany: 40.3 → 41.0

https://ftp.gnome.org/pub/GNOME/sources/epiphany/41/epiphany-41.alpha.news
https://ftp.gnome.org/pub/GNOME/sources/epiphany/41/epiphany-41.beta.news
https://ftp.gnome.org/pub/GNOME/sources/epiphany/41/epiphany-41.rc.news
https://ftp.gnome.org/pub/GNOME/sources/epiphany/41/epiphany-41.0.news

jtojnar committed:
gsettings-desktop-schemas: 40.0 → 41.0

https://ftp.gnome.org/pub/GNOME/sources/gsettings-desktop-schemas/41/gsettings-desktop-schemas-41.alpha.news
https://ftp.gnome.org/pub/GNOME/sources/gsettings-desktop-schemas/41/gsettings-desktop-schemas-41.rc.news
https://ftp.gnome.org/pub/GNOME/sources/gsettings-desktop-schemas/41/gsettings-desktop-schemas-41.0.news

jtojnar committed:
jsonrpc-glib: 3.38.0 → 3.40.0

https://ftp.gnome.org/pub/GNOME/sources/jsonrpc-glib/3.40/jsonrpc-glib-3.40.0.news

jtojnar committed:
sysprof: 3.40.1 → 3.42.0

https://ftp.gnome.org/pub/GNOME/sources/sysprof/3.42/sysprof-3.42.0.news

jtojnar committed:
vte: 0.64.1 → 0.64.2

https://ftp.gnome.org/pub/GNOME/sources/vte/0.64/vte-0.64.2.changes

jtojnar committed:
webkitgtk_4_1: init

4.1 ABI version is the same as 4.0 but it links against libsoup 3.0 ABI.

jtojnar committed:
tracker: 3.1.1 → 3.2.0

https://gitlab.gnome.org/GNOME/tracker/-/blob/3.2.0/NEWS

Notably, both libsoup 2.x and 3.x are linked and selected at runtime (to avoid conflicts).

Also API documentation is now built by hotdoc but the build system copies it from the tarball for now.

jtojnar committed:
evolution-data-server: 3.40.4 → 3.42.0

https://gitlab.gnome.org/GNOME/evolution-data-server/-/blob/3.42.0/NEWS

jtojnar committed:
gnome.gdm: 40.1 → 41.0

jtojnar committed:
gnome.ghex: 3.18.4 → 3.41.0

jtojnar committed:
gnome.gnome-autoar: 0.3.3 → 0.4.0

jtojnar committed:
gnome.gnome-backgrounds: 40.1 → 41.0

jtojnar committed:
gnome.gnome-calculator: 40.1 → 41.0

jtojnar committed:
gnome.gnome-chess: 40.1 → 41.0

jtojnar committed:
gnome.gnome-contacts: 40.0 → 41.0

jtojnar committed:
gnome.gnome-control-center: 40.0 → 41.0

jtojnar committed:
gnome.gnome-disk-utility: 40.2 → 41.0

jtojnar committed:
gnome.gnome-font-viewer: 40.0 → 41.0

jtojnar committed:
gnome.gnome-initial-setup: 40.4 → 41.0

jtojnar committed:
gnome.gnome-maps: 40.4 → 41.0

jtojnar committed:
gnome.gnome-remote-desktop: 40.1 → 41.0

jtojnar committed:
gnome.gnome-settings-daemon: 40.0.1 → 41.0

jtojnar committed:
gnome.gnome-shell: 40.4 → 41.0

jtojnar committed:
gnome.gnome-shell-extensions: 40.4 → 41.0

jtojnar committed:
gnome.gnome-software: 40.4 → 41.0

jtojnar committed:
gnome.gnome-system-monitor: 40.1 → 41.0

jtojnar committed:
gnome.gnome-terminal: 3.40.1 → 3.42.0

jtojnar committed:
gnome.mutter: 40.4 → 41.0

https://gitlab.gnome.org/GNOME/mutter/-/blob/41.0/NEWS

jtojnar committed:
umockdev: 0.15.4 → 0.16.3

jtojnar committed:
nv-codec-headers-11: init at 11.1.5.0

jtojnar committed:
folks: fix build with e-d-s ≥ 3.41

jtojnar committed:
gnome.gpaste: 3.40.2 → 3.42.0

https://www.imagination-land.org/posts/2021-09-28-gpaste-3.42.0-released.html

jtojnar committed:
gnome.hitori: 3.38.2 → 3.38.3

jtojnar committed:
gnome.nautilus: 40.2 → 41.0

jtojnar committed:
gnome.networkmanager-openvpn: 1.8.12 → 1.8.16

jtojnar committed:
gnome.rygel: 0.40.1 → 0.40.2

jtojnar committed:
gnome.seahorse: 40.0 → 41.0

jtojnar committed:
gnome.simple-scan: 40.1 → 40.5

jtojnar committed:
gnome.swell-foop: 40.1 → 41.0.1

jtojnar committed:
gcr: 3.40.0 → 3.41.0

jtojnar committed:
gnome.yelp: 40.3 → 41.1

https://ftp.gnome.org/pub/GNOME/sources/yelp/41/yelp-41.beta.news
https://ftp.gnome.org/pub/GNOME/sources/yelp/41/yelp-41.beta2.news
https://ftp.gnome.org/pub/GNOME/sources/yelp/41/yelp-41.0.news
https://ftp.gnome.org/pub/GNOME/sources/yelp/41/yelp-41.1.news

jtojnar committed:
gnome.yelp-xsl: 40.2 → 41.0

jtojnar committed:
gnome.zenity: 3.32.0 → 3.41.0

https://ftp.gnome.org/pub/GNOME/sources/zenity/3.41/zenity-3.41.0.news

yelp-tools not actually needed, meson uses itstool and gettext:
https://github.com/mesonbuild/meson/blob/3feaea6b29197cd224fbce0ac65fd43d08c3beac/mesonbuild/scripts/yelphelper.py

jtojnar commented:
All packages under `gnome` attribute have been bumped and possible problems noticed in changelogs were added to the [board](https://github.com/NixOS/nixpkgs/projects/41). Probably not ready to test on real systems yet. Feel free to lend a hand in triaging them or submitting fixes. Helping with red jobs on [Hydra](https://hydra.nixos.org/jobset/nixpkgs/gnome) also appreciated.

jtojnar committed:
bolt: fix tests

Newer umockdev pre-creates directories which leads to conflicts when the testsuite tries to create them as well.

Commit:
evolution: 3.40.4 -> 3.42.0

Commit:
evolution-ews: 3.40.4 -> 3.42.0

Commit:
gnome.gnome-user-docs: 40.3 -> 41.0

Commit:
gnome.gucharmap: 13.0.2 -> 14.0.0

SuperSandro2000 commented on pkgs/desktops/gnome/core/gnome-terminal/default.nix:
```diff
@@ -56,11 +62,15 @@ stdenv.mkDerivation rec {
   postPatch = ''
-    substituteInPlace configure --replace '$(eval echo $(eval echo $(eval echo ''${dbusinterfacedir})))/org.gnome.ShellSearchProvider2.xml' "${gnome.gnome-shell}/share/dbus-1/interfaces/org.gnome.ShellSearchProvider2.xml"
-    substituteInPlace src/Makefile.in --replace '$(dbusinterfacedir)/org.gnome.ShellSearchProvider2.xml' "${gnome.gnome-shell}/share/dbus-1/interfaces/org.gnome.ShellSearchProvider2.xml"
+    substituteInPlace src/meson.build --replace "gt_prefix / gt_dbusinterfacedir / 'org.gnome.ShellSearchProvider2.xml'" "'${gnome.gnome-shell}/share/dbus-1/interfaces/org.gnome.ShellSearchProvider2.xml'"
```
```suggestion
    substituteInPlace src/meson.build \
      --replace "gt_prefix / gt_dbusinterfacedir / 'org.gnome.ShellSearchProvider2.xml'" \
      "'${gnome.gnome-shell}/share/dbus-1/interfaces/org.gnome.ShellSearchProvider2.xml'"
```

SuperSandro2000 commented on pkgs/development/libraries/nv-codec-headers/11_x.nix:
```diff
@@ -0,0 +1,29 @@
+
+  meta = with lib; {
+    description = "FFmpeg version of headers for NVENC";
+    homepage = "https://ffmpeg.org/";
```
```suggestion
    homepage = "https://git.videolan.org/?p=ffmpeg/nv-codec-headers.git";
```

SuperSandro2000 commented on pkgs/development/libraries/nv-codec-headers/11_x.nix:
```diff
@@ -0,0 +1,29 @@
+    description = "FFmpeg version of headers for NVENC";
+    homepage = "https://ffmpeg.org/";
+    license = licenses.mit;
+    maintainers = with maintainers; [
+      MP2E
+    ];
```
```suggestion
    maintainers = with maintainers; [ MP2E ];
```

Commit:
atkmm: 2.28.0 -> 2.28.2

- https://gitlab.gnome.org/GNOME/atkmm/-/blob/2.28.2/NEWS

Commit:
atkmm_2_36: init at 2.36.1

- https://gitlab.gnome.org/GNOME/atkmm/-/blob/2.36.1/NEWS

Commit:
atkmm: freeze

Commit:
glibmm: freeze

Commit:
pangomm: freeze

SuperSandro2000 commented on pkgs/development/libraries/atkmm/2.36.nix:
```diff
@@ -0,0 +1,33 @@
+  };
+
+  meta = {
+    description = "C++ wrappers for ATK accessibility toolkit";
```
Missing maintainer

t184256 commented:
Yeah, and probably makes sense to update gnome extensions in the same PR.

jtojnar commented:
Weird, that does not look like something that should fail:

https://gitlab.gnome.org/GNOME/tracker/-/blob/255c935754ca6c0b6a0e05fd411c21c7364ea5cd/tests/libtracker-sparql/tracker-resource-test.c#L61

I was not able to reproduce it so probably something transient.

jtojnar committed:
xdg-desktop-portal-gnome: init at 41.1

jtojnar committed:
nixos/gnome: use GNOME xdg portal

jtojnar committed:
gnomeExtensions.gsconnect: 47 → 48

https://github.com/GSConnect/gnome-shell-extension-gsconnect/releases/tag/v48
https://github.com/GSConnect/gnome-shell-extension-gsconnect/compare/v47...v48

jtojnar committed:
nixos/release-notes: Mention GNOME 41

jtojnar committed:
font-manager: 0.8.6 → 0.8.7

https://github.com/FontManager/font-manager/releases/tag/0.8.7

jtojnar committed:
xdg-desktop-portal: 1.8.1 → 1.10.1

https://github.com/flatpak/xdg-desktop-portal/releases/tag/1.10.0
https://github.com/flatpak/xdg-desktop-portal/releases/tag/1.10.1

jtojnar committed:
xdg-desktop-portal-gtk: 1.8.0 → 1.10.0

https://github.com/flatpak/xdg-desktop-portal-gtk/releases/tag/1.10.0

jtojnar committed:
dee: Fix build with latest Vala

jtojnar commented:
All GNOME apps seem to work, the portal too, failing Hydra packages are pretty low (only few leaf Vala packages) so the only remaining thing is extensions but those can be updated separately when they are released.

Unfortunately, it turns out the tests no longer pass. Likely because the shell api we used is now private: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1970

jtojnar committed:
nixosTests.gnome: fix

GNOME Shell 41 made the Eval command we use for NixOS tests internal
so in order to be still able to use it, we need to run the shell
in an unsafe mode.

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1970
https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1983

Hopefully, the test driver will eventually get better so that
we will not need hacks like this forever.

fabianhjr commented:
Shouldn't this merge request target staging instead of the main branch?

Thanks for working on this. Enjoying GNOME 41 so far and nothing seems to be broken after a whole day of usage(other than some extensions missing upstream GNOME 41 support unu)

jtojnar commented:
Since we build the packages on Hydra, merging to master will not cause a mass rebuild.

t184256 commented:
I somehow still can't figure out a way to rebase this to master and not trigger a huge rebuild, so I'm mildly cautious about that claim.

Running it for 3 days already, still can't find much to complain about
(which kinda undermines my whole reason for trying it: dispelling boredom)
Brightness icon is occasionally volume icon and vice versa when I adjust them, but I seriously doubt this is a packaging issue.

Got any testing guidelines or something?

jtojnar committed:
libshumate: init at unstable-2021-10-06

TredwellGit commented:
`pkgsi686Linux.tracker` fails same as https://github.com/NixOS/nixpkgs/pull/139144#issuecomment-937302301.

maxhille commented:
`nixos-unstable` is now broken for me, too.

```
running tests
ninja: Entering directory `/build/tracker-3.2.0/build'
ninja: no work to do.
 1/33 tracker:functional / fts-functions                      OK               0.89s
 2/33 tracker:functional / sparql-bugs                        OK               0.92s
 3/33 tracker:functional / query                              OK               1.32s
 4/33 tracker:functional / group-concat                       OK               0.68s
 5/33 tracker:functional / coalesce                           OK               0.77s
 6/33 tracker:functional / insertion                          OK               1.84s
 7/33 tracker:functional / distance                           OK               0.73s
 8/33 tracker:functional / graph                              OK               0.92s
 9/33 tracker:functional / collation                          OK               0.70s
10/33 tracker:functional / notifier                           OK               1.02s
11/33 tracker:functional / concurrent-query                   OK               1.49s
12/33 tracker:functional / ontology-rollback                  OK               0.47s
13/33 tracker:functional+ipc / bus-query-cancellation         OK               0.40s
14/33 tracker:gvdb / gvdb                                     OK               0.01s
15/33 tracker:common / file-utils                             OK               0.01s
16/33 tracker:common / utils                                  OK               0.01s
17/33 tracker:data / crc32                                    OK               0.02s
18/33 tracker:data / ontology-change                          OK               0.58s
19/33 tracker:data / ontology-error                           OK               0.03s
20/33 tracker:data / sparql-blank                             OK               0.07s
21/33 tracker:functional / cli                                OK               1.74s
22/33 tracker:data / insert-or-replace                        OK               0.37s
23/33 tracker:functional / ontology-changes                   OK               2.60s
24/33 tracker:functional / portal                             OK               2.89s
25/33 tracker:fts / fts                                       OK               0.39s
26/33 tracker:resource / resource                             FAIL             0.12s   killed by signal 6 SIGABRT
>>> MALLOC_PERTURB_=186 LANG=en_US.UTF8 TRACKER_LANGUAGE_STOP_WORDS_DIR=/build/tracker-3.2.0/src/libtracker-common/stop-words /build/tracker-3.2.0/build/tests/libtracker-sparql/tracker-resource-test
 ✀
stdout:
# random seed: R02S9bb64c7f3a28fd1d4cbc9738781ee4e8
# Testing Tracker resource abstraction
1..6
# Start of libtracker-sparql tests
# Start of tracker-resource tests
ok 1 /libtracker-sparql/tracker-resource/get_empty
Bail out! Tracker:ERROR:../tests/libtracker-sparql/tracker-resource-test.c:61:test_resource_get_set_simple: 'tracker_resource_get_first_double (resource, "http://example.com/1") == 0.6' should be TRUE
stderr:
**
Tracker:ERROR:../tests/libtracker-sparql/tracker-resource-test.c:61:test_resource_get_set_simple: 'tracker_resource_get_first_double (resource, "http://example.com/1") == 0.6' should be TRUE


27/33 tracker:data+slow / ontology                            OK               3.64s
28/33 tracker:data / service                                  OK               5.40s
29/33 tracker:data+slow / sparql                              OK               6.75s
30/33 tracker:sparql / sparql                                 OK               0.44s
31/33 tracker:sparql / statement                              OK               0.57s
32/33 tracker:examples / endpoint                             OK               0.59s
33/33 tracker:sparql / fd                                     OK               0.91s

Summary of Failures:

26/33 tracker:resource / resource                     FAIL             0.12s   killed by signal 6 SIGABRT


Ok:                 32
Expected Fail:      0
Fail:               1
Unexpected Pass:    0
Skipped:            0
Timeout:            0

Full log written to /build/tracker-3.2.0/build/meson-logs/testlog.txt
builder for '/nix/store/d67j41d88js80c6rvk7d6fpap7ilmczd-tracker-3.2.0.drv' failed with exit code 1
cannot build derivation '/nix/store/yw942n07506d9gfkcxc2j0mz83i15j39-gtk+3-3.24.30.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/vh65pnvg2h1rv5gcm6kbrk0y4cwbvv77-steam-usr-multi.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/xv3xrn6fs6w2pyds8dh9hvwcp8vs2sg2-steam-fhs.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/1ivxvf5svrz2k27bgcrz0a5k1hfiwm3d-steam.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/2g1vib9a53hlx3vn5qkgnlpm8c9mpvp5-steam.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/5qyk5w68qbq5cl0adgp05pzwr0ynxgh7-system-path.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/r911pdavjdg67sfpi19cqkab2nkc34qp-nixos-system-venus-21.11.git.7366ad6608f.drv': 1 dependencies couldn't be built
error: build of '/nix/store/r911pdavjdg67sfpi19cqkab2nkc34qp-nixos-system-venus-21.11.git.7366ad6608f.drv' failed
```
